Source/Creation/Frame/Content/Other Frame Creators/ID3ChapterTypeFrameCreator.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  ID3ChapterTypeFrameCreator.swift
+//  ID3TagEditor
+//
+//  Created by Nolaine Crusher on 4/5/20.
+//  Copyright © 2020 Fabrizio Duroni. All rights reserved.
+//
+
+import Foundation
+
+class ID3ChapterTypeFrameCreator: ChapterFrameCreator {
+    
+    private let frameContentSizeCalculator: FrameContentSizeCalculator
+    private let frameFlagsCreator: FrameFlagsCreator
+    private let stringToBytesAdapter: StringToBytesAdapter
+    private let frameCreatorsChain: ID3FrameCreatorsChain
+
+    init(frameContentSizeCalculator: FrameContentSizeCalculator,
+         frameFlagsCreator: FrameFlagsCreator,
+         stringToBytesAdapter: StringToBytesAdapter,
+         frameCreatorsChain: ID3FrameCreatorsChain) {
+        self.frameContentSizeCalculator = frameContentSizeCalculator
+        self.frameFlagsCreator = frameFlagsCreator
+        self.stringToBytesAdapter = stringToBytesAdapter
+        self.frameCreatorsChain = frameCreatorsChain
+    }
+    
+    private func getAsciiCodesOfDigits(_ n: Int)->[UInt8]{
+        return String(n).unicodeScalars.map{UInt8($0.value)}
+    }
+
+    func createFrame(
+        frameIdentifier: [UInt8],
+        version: ID3Version,
+        elementID: String,
+        startTime: Int,
+        endTime: Int,
+        startingByteOffset: Int,
+        endingByteOffset: Int,
+        embeddedSubframes: [FrameName: ID3Frame]) -> [UInt8] {
+        
+        let subframePseudoTag = ID3Tag(
```

Comment:
This is where things get a little hinky. I'm not sure making a "fake" Tag of the subframe is the proper thing to do, but it was the only way I could think of, since the groundwork to convert a frame into bytes is all there.And I am definitely not sure the the `getAsciiCodesOfDigits` function is the right way to convert the Integer frames to bytes.

Reply:
The basic idea seems right to me. But I think it needs to be treated as a sequence of frames instead of a whole tag, because a tag also expects some header information before its frames, right?

Reply:
That had concerned me too, but I spent quite some time looking that over and...I don't think so?

While it's hard to pick things out amidst all the similarly named files, I don't think the header information is added until the tag is actually written to a file (which I think happens in `ID3TagCreator` class.)

All the `ID3Tag` class does is provide the `frames: [FrameName : ID3Frame]` property and the `properties` property which allows you to query size and version for the tag. It doesn't even reference the `ID3TagCreator` class. Instead, `id3Tag: ID3Tag` is a parameter for the `create` method of the `ID3TagCreator` class.

So I think this what I need, except that for some reason, it isn't happening in the test, which is what I'm trying to puzzle out.

I'm not exactly sure the `createFramesUsing` method from `ID3FramesCreator` is the proper one?

```swift
    func createFramesUsing(id3Tag: ID3Tag) throws -> [UInt8] {
        var frames = id3FrameCreatorsChain.createFrames(id3Tag: id3Tag, tag: [UInt8]())
        if thereIsNotValidDataIn(frames: frames) {
            throw ID3TagEditorError.InvalidTagData
        }
        frames.append(contentsOf: createFramesEnd())
        return frames
    }
```

It seems to pull the frames together and check to make sure they're okay, and returns them as [UInt8], which is what I need. `ID3FrameCreatorsChain.createFrames` also returns all the frames as [UInt8] but lacks the layer of validity checking:

```swift
    func createFrames(id3Tag: ID3Tag, tag: [UInt8]) -> [UInt8] {
        if let validNextCreator = nextCreator {
            return validNextCreator.createFrames(id3Tag: id3Tag, tag: tag)
        }
        return tag
    }
```

But I'm not sure if the problem is here in the frame creator, or in the way I'm writing the test. What is coming back from the test is that none of the bytes after `endByteOffset` are being counted.